File: MoneyMoves/MoneyMoves/Theme/Tokens.swift
import SwiftUI

// Soft neo-futurism design tokens
// All colors, gradients, radii, and shadows for Money Moves.

enum Palette {
    static let cream      = Color(hex: 0xFDFBF7)
    static let creamDeep  = Color(hex: 0xF4EFE6)
    static let ink        = Color(hex: 0x1A1726)
    static let inkSoft    = Color(hex: 0x4A4459)
    static let inkMuted   = Color(hex: 0x8B8499)

    static let lavender      = Color(hex: 0xC4B5FD)
    static let lavenderDeep  = Color(hex: 0xA78BFA)
    static let lavenderSoft  = Color(hex: 0xEDE9FE)

    static let peach      = Color(hex: 0xFFD4B8)
    static let peachDeep  = Color(hex: 0xFF9E7D)
    static let peachSoft  = Color(hex: 0xFFF1E6)

    static let mint      = Color(hex: 0xB5E8D5)
    static let mintDeep  = Color(hex: 0x6FE3B3)
    static let mintSoft  = Color(hex: 0xE6FAF1)

    static let rose      = Color(hex: 0xF9C5D1)
    static let roseDeep  = Color(hex: 0xEA9AB2)
    static let roseSoft  = Color(hex: 0xFDECF1)

    static let sky      = Color(hex: 0xBEE3F8)
    static let skyDeep  = Color(hex: 0x7BC4F0)

    static let glass        = Color.white.opacity(0.55)
    static let glassBorder  = Color.white.opacity(0.7)

    // Semantic accents
    static let success = Color(hex: 0x0F8862)
}

enum Gradients {
    static let app = LinearGradient(
        colors: [Palette.cream, Color(hex: 0xF6EEF8), Color(hex: 0xEEF1FA)],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
    static let hero = LinearGradient(
        colors: [Palette.lavender, Palette.rose, Palette.peach],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
    static let mintCard = LinearGradient(
        colors: [Palette.mint, Color(hex: 0xDCF7EB)],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
    static let lavenderCard = LinearGradient(
        colors: [Palette.lavender, Palette.lavenderSoft],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
    static let peachCard = LinearGradient(
        colors: [Palette.rose, Palette.peach],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
}

enum Radius {
    static let sm: CGFloat = 12
    static let md: CGFloat = 18
    static let lg: CGFloat = 24
    static let xl: CGFloat = 32
    static let pill: CGFloat = 999
}

enum Spacing {
    static let xs: CGFloat = 4
    static let sm: CGFloat = 8
    static let md: CGFloat = 12
    static let lg: CGFloat = 16
    static let xl: CGFloat = 24
    static let xxl: CGFloat = 32
    static let xxxl: CGFloat = 48
}

enum Typo {
    static let display     = Font.system(size: 40, weight: .bold).leading(.tight)
    static let h1          = Font.system(size: 30, weight: .bold)
    static let h2          = Font.system(size: 22, weight: .bold)
    static let h3          = Font.system(size: 18, weight: .semibold)
    static let body        = Font.system(size: 15, weight: .regular)
    static let bodyBold    = Font.system(size: 15, weight: .semibold)
    static let caption     = Font.system(size: 12, weight: .medium)
    static let labelSmall  = Font.system(size: 11, weight: .bold).width(.standard)
}

// Hex initializer
extension Color {
    init(hex: UInt32, opacity: Double = 1.0) {
        let r = Double((hex >> 16) & 0xFF) / 255
        let g = Double((hex >> 8) & 0xFF) / 255
        let b = Double(hex & 0xFF) / 255
        self.init(.sRGB, red: r, green: g, blue: b, opacity: opacity)
    }
}

File: src/theme/tokens.ts
export const palette = {
  cream: '#FDFBF7',
  creamDeep: '#F4EFE6',
  ink: '#1A1726',
  inkSoft: '#4A4459',
  inkMuted: '#8B8499',
  white: '#FFFFFF',

  lavender: '#C4B5FD',
  lavenderDeep: '#A78BFA',
  lavenderSoft: '#EDE9FE',

  peach: '#FFD4B8',
  peachDeep: '#FF9E7D',
  peachSoft: '#FFF1E6',

  mint: '#B5E8D5',
  mintDeep: '#6FE3B3',
  mintSoft: '#E6FAF1',

  rose: '#F9C5D1',
  roseDeep: '#EA9AB2',
  roseSoft: '#FDECF1',

  sky: '#BEE3F8',
  skyDeep: '#7BC4F0',
  skySoft: '#E6F4FC',

  shadow: 'rgba(26, 23, 38, 0.08)',
  shadowDeep: 'rgba(26, 23, 38, 0.16)',
  glass: 'rgba(255, 255, 255, 0.55)',
  glassBorder: 'rgba(255, 255, 255, 0.7)',
};

export const gradients = {
  hero: [palette.lavender, palette.rose, palette.peach] as const,
  mint: [palette.mintSoft, palette.mint] as const,
  peach: [palette.peachSoft, palette.peach] as const,
  lavender: [palette.lavenderSoft, palette.lavender] as const,
  rose: [palette.roseSoft, palette.rose] as const,
  sky: [palette.skySoft, palette.sky] as const,
  app: [palette.cream, '#F6EEF8', '#EEF1FA'] as const,
  buddy: ['#FFE5D4', '#E0D4FF', '#D4F0E5'] as const,
};

export const radius = {
  sm: 12,
  md: 18,
  lg: 24,
  xl: 32,
  pill: 999,
};

export const spacing = {
  xs: 4,
  sm: 8,
  md: 12,
  lg: 16,
  xl: 24,
  xxl: 32,
  xxxl: 48,
};

export const typography = {
  display: { fontSize: 40, fontWeight: '700' as const, letterSpacing: -1.2, lineHeight: 44 },
  h1: { fontSize: 30, fontWeight: '700' as const, letterSpacing: -0.6, lineHeight: 36 },
  h2: { fontSize: 22, fontWeight: '700' as const, letterSpacing: -0.3, lineHeight: 28 },
  h3: { fontSize: 18, fontWeight: '600' as const, letterSpacing: -0.2, lineHeight: 24 },
  body: { fontSize: 15, fontWeight: '400' as const, lineHeight: 22 },
  bodyBold: { fontSize: 15, fontWeight: '600' as const, lineHeight: 22 },
  caption: { fontSize: 12, fontWeight: '500' as const, letterSpacing: 0.4, lineHeight: 16 },
  label: { fontSize: 11, fontWeight: '700' as const, letterSpacing: 1.2, lineHeight: 14, textTransform: 'uppercase' as const },
};

export const shadows = {
  soft: {
    shadowColor: palette.ink,
    shadowOffset: { width: 0, height: 8 },
    shadowOpacity: 0.06,
    shadowRadius: 24,
    elevation: 4,
  },
  card: {
    shadowColor: palette.ink,
    shadowOffset: { width: 0, height: 12 },
    shadowOpacity: 0.08,
    shadowRadius: 32,
    elevation: 6,
  },
  glow: {
    shadowColor: palette.lavenderDeep,
    shadowOffset: { width: 0, height: 16 },
    shadowOpacity: 0.25,
    shadowRadius: 36,
    elevation: 10,
  },
};
